tests/st/calicoctl/test_upgrade.py

```diff
@@ -0,0 +1,624 @@
+                            '192.168.214.242/32',
+                            '192.168.215.241/32',
+                            '192.168.216.240/32',
+                            '192.168.217.239/32',
+                            '192.168.218.238/32'],
+             'mac': 'ca:fe:1d:52:bb:e9',
+             'profiles': ['profile1']}
+}
+data['wep_similar_name'] = {
+    'apiVersion': 'v1',
+    'kind': 'workloadEndpoint',
+    'metadata': {'labels': {'app': 'frontend', 'calico/k8s_ns': 'default'},
+                 'name': 'eth0',
+                 'node': 'rack1-host1',
+                 'orchestrator': 'k8s',
+                 'workload': 'default/frontend-5gs43'},
+    'spec': {'interfaceName': 'cali0ef24ba',
+             'ipNetworks': ['192.168.0.0/32',
+                            '192.168.1.255/32',
+                            '192.168.2.254/32',
+                            '192.168.3.253/32',
+                            '192.168.4.252/32',
+                            '192.168.5.251/32',
+                            '192.168.6.250/32',
+                            '192.168.7.249/32',
+                            '192.168.8.248/32',
+                            '192.168.9.247/32',
+                            '192.168.10.246/32',
+                            '192.168.11.245/32',
+                            '192.168.12.244/32',
+                            '192.168.13.243/32',
+                            '192.168.14.242/32',
+                            '192.168.15.241/32',
+                            '192.168.16.240/32',
+                            '192.168.17.239/32',
+                            '192.168.18.238/32'],
+             'mac': 'fe:ed:ca:fe:00:00',
+             'profiles': ['profile1']}
+}
+data['wep_long_labels'] = {
+    'apiVersion': 'v1',
+    'kind': 'workloadEndpoint',
+    'metadata': {'labels': {
+        '8roper/evil-02/key_name.which/is-also_very.long...1234567890//proper/evil-02/key_name.which/is-also_very.long...1234567890//proper/evil-02/Key_Name.which/is-also_very.long...1234567890//proper/evil-02/key_name.which/is-also_very.long...1234567890//proper/evil-02/Key_name.which/is-also_very.long...1234567890//proper/evil-02/key_name.which/is-also_very.long...1234567890//proper/evil-02/key_Name.which/is-also_very.long...1234567890//proper/evil-02/key_name.which/is-also_very.long...1234567890//proper/evil-02/9': 'frontendFrontEnd.0123456789-_-23wdffrontendFrontEnd.0124679-_-0',
+        'calico/k8s_ns': 'default'},
+        'name': '.123Im_a_Little.Teapot-Short_And-Stout.Heres-My-Handle_Heres_My.Spout.Im_AlsoAVeryLongInterfaceNameTryingToCatchOutUpgradeCode75',
+        'node': '.123Im_a_Little.Teapot-Short_And-Stout.Heres-My-Handle_Heres_My.Spout.Im_Also.A.Very.LongNodeNameTryingTo-CatchOut_UpgradeCode75',
+        'orchestrator': 'k8s',
+        'workload': 'default._.123Im_a_Little.Teapot-Short_And-Stout.Heres-My-Handle_Heres_My_AlsoAVeryLongWorkload-NameTryingToCatchOutUpgradeCode5'},
```

Comment:
just confirmed this can't be possible. A workload is made up of `namespace`+`.`+`pod` and k8s pod name follows the following regex `^[a-z0-9]([-a-z0-9]*[a-z0-9])?$`, so it can't begin with a `_`, hence the failure is expected.

Reply:
I don't have a problem with the failure, but the failure message needs to be clear about what the problem is so that the user can fix it.